File: DynamicProgramming1/01Tiles/CS01Tiles/01Tiles.cs
using System;
using System.Collections.Generic;

namespace CS01Tiles
{
    static class Constants
    {
        public const int Mod = 15746;
    }

    class Tiles
    {
        static int n;
        static List<int> d = new List<int>();

        static void GetTiles()
        {
            d.Add(0);
            d.Add(1);
            d.Add(2);

            for (int i = 3; i <= n; i++)
                d.Add((d[i - 1] + d[i - 2]) % Constants.Mod);
        }

        static void Main(string[] args)
        {
            n = Convert.ToInt32(Console.ReadLine());

            GetTiles();

            Console.WriteLine(d[n]);
        }
    }
}

File: DynamicProgramming1/01Tiles/Python01Tiles/01Tiles.py
import sys

MOD = 15746

n = int(sys.stdin.readline())
d = [0, 1, 2]


def tiles():
    for i in range(3, n + 1):
        d.append((d[i - 1] + d[i - 2]) % MOD)


tiles()

print(d[n])
